modules/rostests/apitests/apphelp/env.c

```diff
@@ -686,13 +686,39 @@ static void Test_Shimdata(SDBQUERYRESULT_VISTA* result, const WCHAR* szLayer)
         if (dwSize == sizeof(ShimData_Win2k3))
         {
             /* Fake it for now, so the memcmp works. */
+
             SDBQUERYRESULT_2k3* input = (SDBQUERYRESULT_2k3*)result;
             SDBQUERYRESULT_2k3* output = (SDBQUERYRESULT_2k3*)&result2;
-            ok(output->dwCustomSDBMap == 0, "Expected output->dwCustomSDBMap to be 0, was %u for %s\n", output->dwCustomSDBMap, wine_dbgstr_w(szLayer));
+            const GUID rgGuidDB0 = {0x11111111, 0x1111, 0x1111, {0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11}};
+
+            // Check expected data, for the record.
+            ok(input->dwLayerCount == 1,
+               "Expected input->dwLayerCount to be 1, was %lu for %s\n",
+               input->dwLayerCount, wine_dbgstr_w(szLayer));
+            ok(input->dwCustomSDBMap == 1,
+               "Expected input->dwCustomSDBMap to be 1, was %lu for %s\n",
+               input->dwCustomSDBMap, wine_dbgstr_w(szLayer));
+            ok(IsEqualGUID(&input->rgGuidDB[0], &rgGuidDB0),
+               "Expected input->rgGuidDB[0] to be 11111111-1111-1111-1111111111111111 for %s\n",
+               wine_dbgstr_w(szLayer));
+
+            // Check missing data.
+            ok(output->dwLayerCount == 0,
+               "Expected output->dwLayerCount to be 0, was %lu for %s\n",
+               output->dwLayerCount, wine_dbgstr_w(szLayer));
+            ok(output->dwCustomSDBMap == 0,
+               "Expected output->dwCustomSDBMap to be 0, was %lu for %s\n",
+               output->dwCustomSDBMap, wine_dbgstr_w(szLayer));
+            ok(!memcmp(&output->rgGuidDB[0], &empty_result.rgGuidDB[0], sizeof(empty_result.rgGuidDB[0])),
+               "Expected output->rgGuidDB[0] to be empty for %s\n",
+               wine_dbgstr_w(szLayer));
+
+            // Force new data, for the time being.
+            output->dwLayerCount = input->dwLayerCount;
```

Comment:
I understand this line. What's all the stuff above?

Reply:
Extra-safety checks: to document/check these special TODO cases.